sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -130,18 +170,35 @@ class SQLContext(@transient val sparkContext: SparkContext)
   protected[sql] lazy val optimizer: Optimizer = DefaultOptimizer
 
   @transient
-  protected[sql] val ddlParser = new DDLParser(sqlParser.apply(_))
-
-  @transient
-  protected[sql] val sqlParser = {
-    val fallback = new catalyst.SqlParser
-    new SparkSQLParser(fallback(_))
+  protected[sql] val ddlParser = new DDLParser((sql: String) => { getSQLDialect().parse(sql) })
+  protected[sql] def dialectCache = tlSession.get().dialectCache
```

Comment:
Previously, I ran into timeout during the unittest, as new instance always introduced for each SQL statement. Since we will remove the data `property` from the `HiveDialect`, I will also try to remove the cache here.